openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -286,43 +235,64 @@ pub struct PagePoolOutOfMemory {
     tag: String,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+struct Slot {
+    base_pfn: u64,
+    size_pages: u64,
+    state: SlotState,
```

Comment:
This slot thing gets even simpler if it's `Option<SlotState>`, because then device_id and tag come out, too.

Reply:
Oh, that's a good idea.

Reply:
Doesn't quite work with the upcoming change i have - if we don't save device_ids, then we have different types for device_id depending on if you are a pending restore or a linked allocation with a handle.